enhancements/installer/signal-cluster-deletion.md

```diff
@@ -0,0 +1,152 @@
+---
+title: signal-cluster-deletion
+authors:
+  - "@abutcher"
+reviewers:
+  - dgoodwin
+  - abhinavdahiya
+  - wking
+  - eparis
+  - ecordell
+
+approvers:
+  - dgoodwin
+  - abhinavdahiya
+  - wking
+  - eparis
+creation-date: 2020-03-30
+last-updated: 2020-05-07
+status: provisional
+---
+
+# Signal Cluster Deletion
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Operators can create external resources in clouds and other services that have no opportunity to be cleaned up when a cluster is uninstalled, causing extra work for cluster maintainers to track down and remove these resources. Signalling cluster deletion would allow operators to clean up external resources during uninstallation and would ensure that total cleanup is an automatic process.
+
+## Motivation
+
+### Goals
+
+Signal that cluster deletion has begun through deletion of a new CRD `Alive` and wait for that object to be deleted during uninstall before continuing with removing cluster resources such as instances, storage and networking.
+
+### Non-Goals
+
+* Attaching finalizers to operator resources based on the `Alive` object which would facilitate removal of operator resources.
+* Removing resources other than the `Alive` object during cluster uninstall.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As an OpenShift administrator, I would like for any external resources created in service of the cluster be removed when when I delete the cluster, so that:
+* I do not have to track down those resources and remove them manually
+* I do not have to pay for resources that are not being used.
+
+#### Story 2
+As an OpenShift (software, big-O) Operator, I would like to have a way to know that the cluster is being shut down, so that I can perform any necessary cleanup steps and remove external resources.
+
+#### Story 3
+As an OpenShift administrator, I would like to have a way to indicate that certain external resources that would normally be cleaned up on cluster deletion be preserved on cluster delete, so that I don't lose important resources stored externally.
```

Comment:
This could easily be a foot gun.  If the external resource is incorrectly marked and holds customer data, it could be quite bad.

Reply:
I agree with @jim-minter here, we should not rely on label and artifact markup outside the cluster - each resource incluster, should track the artifacts it needs to / wants to lifecycle and our experience should optimise for that.

Reply:
Agree. The interface for preserving anything off cluster should be a CR owned by the operator that created the off cluster resource(s).